### PR TITLE
[spaceship] Add retrieve-create of iap shared secret to Spaceship::Tunes

### PR DESCRIPTION
**Pull request by piechart**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [ ] I've updated the documentation if necessary.

### Motivation and Context
I was working on iOS project in a remote team. Once someone was implementing in-app purchases functionality and regenerated shared secret key. Unluckily, at the same time I was testing other iaps in the app. Having fastlane as a helping tool which lanes prevent a lot of other cases of such scenarios in the project, I thought being able to retrieve actual shared secret via Spaceship and set in the project somehow would be a useful feature.

### Description
Retrieve currently existing shared secret (or create if does not exist)
```ruby
Spaceship::Tunes.login
app = Spaceship::Tunes::Application.find('bundle_identifier')
secret = app.in_app_purchases.get_shared_secret(create: true)
```

Generate (or regenerate if exists) new app-specific shared secret
```ruby
Spaceship::Tunes.login
app = Spaceship::Tunes::Application.find('bundle_identifier')
secret = app.in_app_purchases.generate_shared_secret
```

### CODE REVIEW

**Comment by piechart** [comment]
@googlebot I signed it!

File: spaceship/lib/spaceship/tunes/tunes_client.rb
Comment (joshdholtz) [inline spaceship/lib/spaceship/tunes/tunes_client.rb:1431]: Can we actually split this into two methods? One for the `:get` and one for the `:post`. I’d just like this to follow the rest of the conventions in this class and keep the `:get` and `:post` encapsulated into this file 😇

**Review by joshdholtz — CHANGES_REQUESTED** [review]
I had no idea this was a thing 🙃 Thanks for the PR for this! I requested one little change to split the tunes client method into two methods. Would you also be able to add some tests for this as well?

**Comment by joshdholtz** [comment]
@max-ott Can you take a look at these changes too? 😊

File: spaceship/lib/spaceship/tunes/tunes_client.rb
Comment (piechart) [reply spaceship/lib/spaceship/tunes/tunes_client.rb]: Thanks for review, totally agree. Have done this! 👍

**Review by joshdholtz — APPROVED** [review]
Looks great to me! Thanks for making those changes ❤️ I’ll wait for @max-ott’s review before merging 😊

**Comment by max-ott** [comment]
Works like a charm! Thanks for the addition.

**Review by max-ott — APPROVED** [review]
Tested and everything is working as intended!

**Comment by piechart** [comment]
@max-ott thank you for review!
@joshdholtz seems like it is ready to be merged now 😃

**Pull request merged**
